### PR TITLE
Issue 162 Add user interaction for stop command

### PR DESCRIPTION
**Pull request by praveenkumar**

### CODE REVIEW

**Comment by anjannath** [comment]
So this means in ci we should always use `crc stop -f`

File: cmd/crc/cmd/stop.go [hunk 1]
```diff
@@ -22,15 +26,46 @@ var stopCmd = &cobra.Command{
...
+func runStop(arguments []string) {
 	stopConfig := machine.StopConfig{
 		Name: constants.DefaultName,
 	}
 
+	killConfig := machine.KillConfig{
+		Name: constants.DefaultName,
+	}
+
+	if force {
```
Comment (anjannath) [inline cmd/crc/cmd/stop.go]: :+1:

File: cmd/crc/cmd/stop.go [hunk 2]
```diff
@@ -22,15 +26,46 @@ var stopCmd = &cobra.Command{
...
+		killVM(killConfig)
+		errors.Exit(0)
+	}
+
 	commandResult, err := machine.Stop(stopConfig)
+	if err != nil {
+		// Here we are catching the error "VM Failed to gracefully shutdown, try the kill command"
+		// from the driver side and if it exist then we ask user input for power off operation.
+		if strings.Contains(err.Error(), "kill") {
```
Comment (anjannath) [inline cmd/crc/cmd/stop.go]: :+1:

File: cmd/crc/cmd/stop.go [hunk 3]
```diff
@@ -22,15 +26,46 @@ var stopCmd = &cobra.Command{
...
+				errors.Exit(0)
+			}
+		}
+		errors.ExitWithMessage(1, err.Error())
+	}
+	output.Out(commandResult.Success)
+}
+
+func killVM(killConfig machine.KillConfig) {
+	commandResult, err := machine.Kill(killConfig)
```
Comment (gbraad) [inline cmd/crc/cmd/stop.go]: Isn't this a `PowerOff` ? Kill sounds more like a `kill -9` of the VM process.

File: pkg/crc/machine/machine.go [hunk 4]
```diff
@@ -152,6 +152,28 @@ func Stop(stopConfig StopConfig) (StopResult, error) {
...
+	libMachineAPIClient := libmachine.NewClient(constants.MachineBaseDir, constants.MachineCertsDir)
+	host, err := libMachineAPIClient.Load(killConfig.Name)
+
+	if err != nil {
+		result.Success = false
+		result.Error = err.Error()
+		return *result, err
+	}
+
+	if err := host.Kill(); err != nil {
```
Comment (gbraad) [inline pkg/crc/machine/machine.go]: this naming is not given by us, so I do not care much.

**Comment by gbraad** [comment]
it is not really 'user interaction' but introducing a 'force' option. title/description does not match here with expectation

File: cmd/crc/cmd/stop.go (hunk 2 above)
Comment (gbraad) [reply cmd/crc/cmd/stop.go]: do we need to do this? the result is a `false`. can't we use this instead?
t he state is running and returned false on stop.,,

besides, this error message needs to be modified in the driver, which means tis check is fragile.

File: cmd/crc/cmd/stop.go (hunk 3 above)
Comment (jsliacan) [reply cmd/crc/cmd/stop.go]: Also, if the user chooses `n`, then it defaults to the message `[VM Failed to gracefully shutdown, try the kill command]` even though in the meantime the VM did shut down gracefully.

File: cmd/crc/cmd/stop.go (hunk 2 above)
Comment (praveenkumar) [reply cmd/crc/cmd/stop.go]: Problem is the state, if your VM is in `stop` or `delete` state then error will be returned and user have to prompted for it which doesn't make sense :(

File: cmd/crc/cmd/stop.go (hunk 3 above)
Comment (praveenkumar) [reply cmd/crc/cmd/stop.go]: That is what driver is returning `VM Failed to gracefully shutdown, try the kill command`, I am still think that just have `-f` for stop no user interaction at all, we will increase the timelimit around `120sec` on driver side.

**Comment by praveenkumar** [comment]
@gbraad updated as per your comments.

**Comment by anjannath** [comment]
It'd bee good to have this in as it unblocks #121

File: cmd/crc/cmd/stop.go (hunk 3 above)
Comment (gbraad) [reply cmd/crc/cmd/stop.go]: I saw you made those changes

File: cmd/crc/cmd/stop.go (hunk 2 above)
Comment (gbraad) [reply cmd/crc/cmd/stop.go]: This got changed by additionally checking for state == running.

**Pull request merged**